core/imds/src/main/java/software/amazon/awssdk/imds/MetadataResponse.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.imds;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.protocols.jsoncore.JsonNode;
+import software.amazon.awssdk.protocols.jsoncore.JsonNodeParser;
+
+/**
+ * The class is used for Response Handling and Parsing the metadata fetched by the get call in the {@link Ec2Metadata} interface.
+ * Metadata is stored in the instance variable <b>body</b>. The class provides convenience methods to the users to parse the
+ * metadata as a String, List and to parse the metadata in the JsonResponse according to the key.
+ */
+@SdkPublicApi
+public class MetadataResponse {
+
+    private static final Logger log = LoggerFactory.getLogger(MetadataResponse.class);
+
+    private static final JsonNodeParser JSON_NODE_PARSER = JsonNode.parser();
```

Comment:
We probably want removeErrorLocations to be true on this parser, so that if credentials are in the metadata it won't include them in the exception message (which might be logged).